gem/node.py

```diff
@@ -99,6 +100,13 @@ def get_hash(self):
         return hash((type(self),) + self._cons_args(self.children))
 
 
+def _make_traversal_children(node):
+    if isinstance(node, (gem.Indexed, gem.FlexiblyIndexed)):
+        return node.children + node.indirect_children
+    else:
+        return node.children
```

Comment:
Note: I'm not sure there is necessarily a better way of doing what you're trying to do, but the fact that the `indirect_children` must be handled specially in every traversal is slightly concerning. It's very easy to forget to do it, I would think.

Reply:
I think the issue is that there is a gap between the primal DAG and the DAG inside `VariableIndex`, as `children` of `Indexed` object does not include `multiindex`.

I at least added some comments in the docstrings.